lib/tasks/support/recovery_db/partial_migration.rb

```diff
@@ -0,0 +1,617 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+
+module RecoveryDB
+  module PartialMigration
+    class RestoreUsers
+      attr_reader :user_ids, :file
+
+      def initialize(user_ids: nil, file: nil)
+        @user_ids = user_ids
+        @file = file
+        mount.restore_backup
+        mount.load_models
+      end
+
+      def mount
+        @mount ||= RecoveryDB::Mount.new(file: @file,
+                                         tables: %w[users profiles collections samples reactions wellplates wells
+                                                    screens research_plans device_descriptions attachments
+                                                    containers container_hierarchies collections_samples
+                                                    collections_reactions collections_wellplates collections_screens
+                                                    collections_research_plans collections_device_descriptions
+                                                    sync_collections_users users_devices devices collections_elements
+                                                    elements element_klasses elements_elements elements_samples
+                                                    segment_klasses segments user_labels users_admins users_groups])
+      end
+
+      def run
+        raise ArgumentError, 'User Ids are required' if @user_ids.nil?
+        raise ActiveRecord::RecordNotFound, 'No user found' if rec_users.empty?
+
+        user_id_map = {}
+        created_elements = {}
+
+        @mount.log_event "Restoring users: found #{rec_users.size} of #{@user_ids.size} user(s) to restore"
+        restore_element_klasses(created_elements)
+        rec_users.each do |recovery_user|
+          new_user = restore_user(recovery_user, created_elements)
```

Comment:
-  Many restore_* methods run DB operations without wrapping them in transactions.
- If part of a restore fails, you may end up with half-restored, inconsistent state.
- Suggestion: Use transactions at critical levels:
```ruby
ActiveRecord::Base.transaction do
  new_user = restore_user(...)
  restore_profile(...)
  restore_collections(...)
end
```